src/Algebra/Definitions/RawMonoid.agda

```diff
@@ -65,3 +66,18 @@ suc n ×′ x = n ×′ x + x
 
 sum : ∀ {n} → Vector Carrier n → Carrier
 sum = Vector.foldr _+_ 0#
+
+------------------------------------------------------------------------
+-- 'Conjunction' with a Boolean: action of the Boolean (true,∧)-rawMonoid
+------------------------------------------------------------------------
+
+infixr 8 _∧_
+
+_∧_ : Bool → Carrier → Carrier
+b ∧ x = if b then x else 0#
+
+-- tail-recursive optimisation
```

Comment:
optimisation of *what*? It doesn't seem like an optimisation of the above? What am I missing?

Reply:
As with the TCOptimised forms of (monoid) sum and multiplication, the 'optimisation' lies in avoiding having to explicitly track terms of the form `x + 0#` when these can be computed away instead to `x`... But I admit I haven't *quite* got my story straight yet... ;-)

Reply:
As witnessed by `b∧x∙y≈b∧x+y`...

Reply:
It still feels like a stretch. I do like the optimization of an action not doing a (statically known) null operation, a lot. I'm still worried that this particular case is a coincidence rather than fundamental!

Reply:
Well I think it extends to the `Nat` cases as well (but obscured in the current presentation, which doesn't introduce the tail-recursive forms...)